### PR TITLE
Open telemetry

### PR DESCRIPTION
**Pull request by brandynbennett**
Adds OpenTelemetry tracing so that published and received messages will be connected in the same trace.

Correlating Protocol PR https://github.com/SpiffInc/polyn-protocol/pull/21

Also fixes a bug where the source_root might get duplicated

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -30,3 +30,9 @@ gem "timecop"
 gem "json_schemer"
 # EventMachine nats repo doesn't support jetstream, only nats-pure
 gem "nats-pure", "~> 2.0"
+gem "opentelemetry-api", "~> 1.1"
+
+group :test do
+  gem "opentelemetry-sdk", "~> 1.2"
```
Comment (xathien) [inline Gemfile:36]: Should I be weirded out that these two versions are different? 😄

File: lib/polyn/event.rb [hunk 2]
```diff
@@ -93,23 +92,27 @@ def to_h
...
     end
 
     ##
     # Get the Event `source` prefixed with reverse domain name
     def self.full_source(source = nil)
-      root = Polyn.configuration.source_root
-      name = Polyn::Naming.dot_to_colon("#{domain}:#{root}")
+      root    = Polyn.configuration.source_root
+      parts   = [domain, root]
+      combine = lambda do |items|
```
Comment (xathien) [inline lib/polyn/event.rb:104]: Huh. I thought Rubocop yelled at us when we used `lambda do` instead of `-> (items) {`

**Review by xathien — APPROVED** [review]

File: Gemfile (hunk 1 above)
Comment (brandynbennett) [reply Gemfile:36]: They are different gems technically so I guess that's ok?

File: lib/polyn/event.rb (hunk 2 above)
Comment (brandynbennett) [reply lib/polyn/event.rb:104]: It yelled if I did a single line with `lambda`, but if it was multi-line it insisted I use `lambda`

Comment (xathien) [reply lib/polyn/event.rb:104]: Ugh. Thanks, Rubocop, for your helpful consistency. 🙄

**Pull request merged**